core/src/value.rs

```diff
@@ -210,7 +212,75 @@ pub enum PrimitiveValue {
     Time(C<NaiveTime>),
 }
 
+/// A utility macro for implementing the conversion from a core type into a
+/// DICOM primitive value with a single element.
+macro_rules! impl_from_for_primitive {
+    ($typ: ty, $variant: ident) => {
+        impl From<$typ> for PrimitiveValue {
+            fn from(value: $typ) -> Self {
+                PrimitiveValue::$variant(C::from_elem(value, 1))
+            }
+        }
+    };
+}
+
+impl_from_for_primitive!(u8, U8);
+impl_from_for_primitive!(u16, U16);
+impl_from_for_primitive!(i16, I16);
+impl_from_for_primitive!(u32, U32);
+impl_from_for_primitive!(i32, I32);
+impl_from_for_primitive!(u64, U64);
+impl_from_for_primitive!(i64, I64);
+impl_from_for_primitive!(f32, F32);
+impl_from_for_primitive!(f64, F64);
+
+impl_from_for_primitive!(Tag, Tags);
+impl_from_for_primitive!(NaiveDate, Date);
+impl_from_for_primitive!(NaiveTime, Time);
+impl_from_for_primitive!(DateTime<FixedOffset>, DateTime);
+
+/// Construct a DICOM value.
+#[macro_export]
+macro_rules! dicom_value {
+    ($typ: ident, [ $($elem: expr),* ]) => {
+        {
+            use smallvec::smallvec; // import smallvec macro
+            dicom_core::value::PrimitiveValue :: $typ (smallvec![$($elem,)*])
+        }
+    };
+    ($typ: ident, $elem: expr) => {
+        dicom_core::value::PrimitiveValue :: $typ (dicom_core::value::C::from_elem($elem, 1))
+    };
+}
+
+impl From<String> for PrimitiveValue {
+    fn from(value: String) -> Self {
+        PrimitiveValue::Str(value)
+    }
+}
+
+impl From<&str> for PrimitiveValue {
+    fn from(value: &str) -> Self {
+        PrimitiveValue::Str(value.to_owned())
+    }
+}
+
 impl PrimitiveValue {
+    /// Create a single unsigned 16-bit value.
+    pub fn new_u16(value: u16) -> Self {
+        PrimitiveValue::U16(C::from_elem(value, 1))
+    }
+
+    /// Create a single unsinged 32 value.
```

Comment:
The most unsung kind of typo.

```suggestion
    /// Create a single unsigned 32-bit value.
```